### PR TITLE
ANA-1090: Cross-currency, basis, and amortising swap examples

### PR DESCRIPTION
**Pull request by tfnc-finbourne**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 1]
```diff
@@ -281,14 +272,13 @@ internal static InterestRateSwap CreateSwapByNamedConventions(DateTimeOffset? st
             return irs;
         }
 
-        // Valid types are "Vanilla"
         internal static InterestRateSwap CreateExampleInterestRateSwap(IRSTypes type)
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:275]: Not your code but I think I'd argue `InterestRateSwapType` (not abbrev'd and not plural)

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            Assert.That(upsertResponse.Failed.Count, Is.EqualTo(0));
+            Assert.That(upsertResponse.Values.Count, Is.EqualTo(quoteRequests.Count));
+
+            Dictionary<string, UpsertComplexMarketDataRequest> upsertComplexMarketDataRequest = new Dictionary<string, UpsertComplexMarketDataRequest>();;
+            UpsertStructuredDataResponse upsertComplexMarketDataResponse;
+            // For models requiring discount curves, we upsert them below. ConstantTimeValueOfMoney does not require any discount curves. 
+            if (model != ModelSelection.ModelEnum.ConstantTimeValueOfMoney)
+            {
+                // provide discount/projection curves in USD, for swaps with a 3M floating leg
+                upsertComplexMarketDataRequest.Add("discount_curve_USD",
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs:61]: You previously argued to "upsert the minimal set". Can this be done here (it looks like you're up'ing all curves)

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            
+            // DELETE instrument
+            _instrumentsApi.DeleteInstrument("ClientInternal", uniqueId);
+        }
+       
+        [LusidFeature("F22-52")]
+        [TestCase(InstrumentExamples.IRSTypes.Basis)]
+        [TestCase(InstrumentExamples.IRSTypes.CrossCurrency)]
+        [TestCase(InstrumentExamples.IRSTypes.Amortising)]
+        public void InterestRateSwapInlineValuationExample(InstrumentExamples.IRSTypes irsType)
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs:144]: How about non-inline valuation?

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 4]
```diff
@@ -281,14 +272,13 @@ internal static InterestRateSwap CreateSwapByNamedConventions(DateTimeOffset? st
...
         {
             InterestRateSwap swap = null;
             switch (type)
             {
                 case IRSTypes.Vanilla:
-                    swap = CreateInterestRateSwap(
+                    swap = CreateInterestRateSwapForIndex(
```
[the remark this answers is not in the document]
Comment (tfnc-finbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:281]: Example takes one parameter, the highlighted method is basically fully configurable. Hence I would argue that it's not an "example" method, it's a literal IRS constructor

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs (hunk 1 above)
Comment (tfnc-finbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs]: done :shrug:

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs (hunk 2 above)
Comment (tfnc-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs:66]: bit messy but done

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs (hunk 3 above)
Comment (tfnc-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs]: added

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapRfRExamples.cs [hunk 5]
```diff
@@ -157,24 +157,24 @@ public void InterestRateSwapRfRCreationAndUpsertionExample(InstrumentExamples.IR
...
-        [TestCase(InstrumentExamples.IRSTypes.SONIA)]
-        [TestCase(InstrumentExamples.IRSTypes.TONA)]
-        [TestCase(InstrumentExamples.IRSTypes.SARON)]
-        public void InterestRateSwapRfRPortfolioCashFlowsExample(InstrumentExamples.IRSTypes rfr)
+        [TestCase(InstrumentExamples.InterestRateSwapType.SOFR)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.ESTR)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.SONIA)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.TONA)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.SARON)]
+        public void InterestRateSwapRfRPortfolioCashFlowsExample(InstrumentExamples.InterestRateSwapType rfr)
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapRfRExamples.cs:177]: Discounting and CTVOM plz

Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapRfRExamples.cs:177]: Not your code, lolz

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        [LusidFeature("F22-54")]
+        [TestCase(InstrumentExamples.InterestRateSwapType.Basis)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.CrossCurrency)]
+        [TestCase(InstrumentExamples.InterestRateSwapType.Amortising)]
+        public void InterestRateSwapPortfolioCashFlowsExample(InstrumentExamples.InterestRateSwapType interestRateSwapType)
+        {
+            var irs = InstrumentExamples.CreateExampleInterestRateSwap(interestRateSwapType);
+            CallLusidGetPortfolioCashFlowsEndpoint(irs, ModelSelection.ModelEnum.Discounting);
+        }
+
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs:179]: pedantic: blank line
and L168

**Review by rizwansaeed — APPROVED** [review]

**Pull request merged**